lib/spack/spack/analyzers/environment_variables.py

```diff
@@ -43,6 +44,7 @@ def _read_environment_file(self, filename):
         to remove path prefixes specific to user systems.
         """
         if not os.path.exists(filename):
+            tty.warn("No environment file available")
             return
```

Comment:
Returning None here will  be passed into the results dict and then not trigger the if statement here: https://github.com/spack/spack/blob/b83f06df0c4262afd43bf8816bae0db84247f849/lib/spack/spack/analyzers/analyzer_base.py#L99. So that should be good!